src/components/shared/qrcode.tsx

```diff
@@ -31,9 +31,11 @@ export function QRCode({ text, options = {}, canvasProps }: QRProps) {
     useEffect(() => {
         if (!ref.current) return
 
-        qr.toCanvas(ref.current, text, options, () => {
-            if (ref.current?.toDataURL()) {
-                cache.set(text, ref.current?.toDataURL())
+        qr.toCanvas(ref.current, text, options, (err: Error) => {
+            if (err) throw err
+            const dataURL = ref.current?.toDataURL()
+            if (dataURL) {
+                cache.set(text, dataURL)
             }
         })
     }, [options, text])
```

Comment:
options👀

Reply:
You mean should the options also be cached?